UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs

```diff
@@ -149,7 +160,8 @@ protected struct GLBStream
 		protected ILoader _loader;
 		protected bool _isRunning = false;
 
-
+		public ImportProgress Progress { get; } = new ImportProgress();
```

Comment:
Instead of a public Progress, you should:

Take in an optional IProgress<ImportProgress> progress in LoadSceneAsync() and use that to report progress.

Then the ImportProgress should probably also change to make all of its public fields become public getter-only properties that are set in the constructor.  So you construct one then call IProgress.Report();

Or make ImportProgress a struct so that it is passed by value.  Basically I don't want it to be possible for the caller to accidentally (or maliciously) set the progress value on us.

Reply:
See new commit for details